### ISSUE TITLE
Switch to ESLint (?)

### ISSUE THREAD

**Issue by tripu** [w3c/echidna#215]
&hellip;which has a plugin architecture, and supposedly could do what JSCS and JSHint do now, but faster.

I realise that claim would have to be evaluated first, and that our current style rules would need to be rewritten. So maybe not worth it. This issue is just a reminder.

[`github.com/eslint/eslint`](https://github.com/eslint/eslint)

**Comment by astorije** [comment]
Well, I'm not sure it's worth spending time on that. I think you need to ask yourself (a) is there something that new tool offers, we need it and cannot have it with that current tool? (b) what is the ratio time needed / value added with that switch?
Considering JSCS runs (on my machine) in .9s and JSHint runs in .8s, I'm not sure improving the velocity of the tool should be considered.

If we had none already in place, I agree we would definitely consider both solutions. I just chose this one on personal time one night and actually spent the night working on the rules and cleaning up the existing, finding several hidden bugs in the process.
This takes more time that one expects, and in the end I am happy I didn't spend time on comparing solutions. Given the size of the team and the tool, any study of more than 10 minutes would have been a waste of time. Now there is a solution in place, and that prevents easy mistakes from devs and contributors, better than nothing, and nice to keep a consistent style over the project without wasting time on this style, also known as the biggest bikeshed in computer engineering :-)

(I essentially agree that JSHint is not a great tool to use and its separation from JSCS seems odd, but oh well, that doesn't prevent me to sleep at night :p)

**Comment by tripu** [comment]
Yes, so I thought. Unless ESLint gives us significantly more, and/or the switch is immediate, I wouldn't touch this now, @astorije.

**Comment by IanVS** [comment]
I'd say the biggest benefits behind ESLint is that you can write your own rules if you need to, and it has great support for ES6 (and even some ES7).  There is also https://github.com/feross/standard which can help prevent a lot of bikeshedding, especially if the rules already line up well with what you're using.  

One feature we are working on inside of ESLint is auto-config, which will examine your code and choose a rule configuration which most closely matches what you're already doing.  I'd suggest keeping an eye out for that, which will give an easy way to switch.  If you've got any other questions about ESLint, I'd be happy to try to answer them.

**Comment by tripu** [comment]
Thanks for those pointers, @IanVS! It's good to have an opinion first-hand (if also biased ;)

**Comment by IanVS** [comment]
Definitely biased.  :wink:   If what you've got going is working for you, keep using it for sure.
